### DISCUSSION TITLE
Can you gracefully filter on a custom multivalued person field without a custom SPFx component?

### DISCUSSION THREAD

**Discussion by ekdegerman** [microsoft-search/pnp-modern-search#1771] [Q&A]
We are using the PnP modern search web parts to search and filter content in a SharePoint Online based application that contains a large list (above the threshold, meaning the OOB SharePoint list search starts limiting the number of matches it returns). One of the fields we would like to filter on is a multivalued person field. Based on among other things previous issues and posts in this repository we have tried a number of different approaches but can't really seem to get all the way to the desirable behavior, at least using only configuration. What we want is something that behaves like filtering on the OOB DisplayAuthor field, where each individual author name is its own filter rather than every combination of people added to an item. What we have tried so far:

**List or site column with the clear text name crawled property (e.g. ows_MultiPeopleSite) mapped to a RefinableString managed property**
While this will give us the display names of the authors, it creates refiner values with all the combinations of people assigned to items rather than the individual names, leading to e.g. refiner values like the ones below: 

- User One
- User One;User Two
- User Two
- User Two; User One

I know this flawed filter multi value handling is coming from the SharePoint search engine itself and is not specific to PnP modern search, the search query tool shows the same combined values. It doesn't make for a great experience when filtering, though.

**Site column with the USER crawled property (e.g. ows_q_USER_MultiPeopleSite) mapped to a RefinableString managed property**
This approach splits the filter values per author like we want, and with a custom layout template we can clean up the labels and show only the name. However, it seems refining on those values do not work in SharePoint Online. Whether in the PnP Search Filters web part or in a classic search center created for testing purposes, any choice just returns 0 results. It also seems that, unlike in SharePoint 2016 where such filters work at least with other managed properties, the hex value in the token doesn't quite match its text value, not sure if that could be the reason. We did also try rewriting the value in the layout template to use the e-mail field instead of the value. With this approach I can filter successfully and get the correct result, but since what I filtered on does not match any of the original refiners I had, I get an extra entry in the dropdown with no count. I'm a real beginner when it comes to handlebars so I don't know if I missed something obvious in this last case that would allow us to fix it, but so far neither option using this property is great either. Below is the change I added to the rendering of the options if that helps...

```html
{{#if (equalsLength (split name '|') 1)}}
	<option value="{{value}}" data-selected="{{selected}}" data-disabled="{{disabled}}" data-count="{{count}}">{{name}}</option>
{{else}}
	<option value="{{#with (split name '|')}}{{[0]}}{{/with}}" data-selected="{{selected}}" data-disabled="{{disabled}}" data-count="{{count}}">{{#with (split name '|')}}{{[1]}}{{/with}}</option>
{{/if}}
```

I also saw that there used to be a persona filter template back in version 3, but that it is no longer available in version 4. Was it decommissioned due to similar issues or are there any plans to release one in the new version as well at some point? Any other recommendations for how to get around these limitations using configuration only or would custom components be required to create a workaround?

**Comment by wobba** [comment]
You probably need to add logic to a custom filter template where you loop over the values coming back and see if you can split and re-group them so that "User One; User Two" is merged with "User One" and "User Two" which has single values.

And then use this back in say a `pnp-filtercheckbox`. Make a note that I use `name` for both the name and value part. As you don't need to send the encoded value back. It's just FQL.

```
<pnp-filtercheckbox 
				data-filter-name="{{filter.filterName}}" 
				data-name="{{name}}" 
				data-value="{{name}}" 
				data-selected="{{selected}}" 
				data-disabled="{{disabled}}"
				data-theme-variant="{{JSONstringify @root.theme}}"
			>
			</pnp-filtercheckbox>
```

Probably easier to have a custom web component do the work, or a custom handlebar helper method to re-work the object structure coming back. Something which would be useful in the web parts as a default offering probably.

**Reply by ekdegerman** [reply]
Thanks for a quick response! Having such a template available in the default offering would definitely be helpful! :) 
So, in your opinion it is better in general to use and manipulate the property containing the displaynames of the users than the one containing full user info? One benefit of that in our scenario would be that we don't have to go to site columns...

**Reply by wobba** [reply]
Not saying one is better.. it depends on your scenario :)

**Reply by ekdegerman** [reply]
For our scenario list fields are definitely preferable. :-) If you were to guess, what are the chances that you would incorporate support for this in the default offering within the next few months?

**Reply by wobba** [reply]
I have no idea... personally I don't have much time for development these days. If you can enlist someone to help out with a pull request, we would take it :) There is some complexity such as how to present counts correctly, and then also ensure the click is handled the right way.
